File: C++/1314.cpp
#include <iostream>
using namespace std;


int main()
{
    int length1, width1, height1;
    int length2, width2, height2;
    int lengthC, widthC, heightC;

    cin >> length1 >> width1 >> height1
        >> length2 >> width2 >> height2
        >> lengthC >> widthC >> heightC;

    if (height1 > heightC || (width1 > widthC && width1 > lengthC)
      || (length1 > widthC && length1 > lengthC)
      || height2 > heightC || (width2 > widthC && width2 > lengthC)
      || (length2 > widthC && length2 > lengthC))
    {
        cout << "NO" << endl;
    }
    else if ((height1 + height2 <= heightC)
      || (length1 + length2 <= lengthC && width1 <= widthC && width2 <= widthC)
      || (length1 + width2 <= lengthC && width1 <= widthC && length2 <= widthC)
      || (width1 + length2 <= lengthC && length1 <= widthC && width2 <= widthC)
      || (width1 + width2 <= lengthC && length1 <= widthC && length2 <= widthC)
      || (length1 + length2 <= widthC && width1 <= lengthC && width2 <= lengthC)
      || (length1 + width2 <= widthC && width1 <= lengthC && length2 <= lengthC)
      || (width1 + length2 <= widthC && length1 <= lengthC && width2 <= lengthC)
      || (width1 + width2 <= widthC && length1 <= lengthC && length2 <= lengthC))
    {
        cout << "YES" << endl;
    }
    else
    {
        cout << "NO" << endl;
    }

    return 0;
}

File: Python/1314.py
l1, w1, h1 = map(int, input().split())
l2, w2, h2 = map(int, input().split())
lc, wc, hc = map(int, input().split())

if h1 > hc or (w1 > wc and w1 > lc) or (l1 > wc and l1 > lc)\
  or h2 > hc or (w2 > wc and w2 > lc) or (l2 > wc and l2 > lc):
    print('NO')
elif h1 + h2 <= hc\
  or l1 + l2 <= lc and w1 <= wc and w2 <= wc\
  or l1 + w2 <= lc and w1 <= wc and l2 <= wc\
  or w1 + l2 <= lc and l1 <= wc and w2 <= wc\
  or w1 + w2 <= lc and l1 <= wc and l2 <= wc\
  or l1 + l2 <= wc and w1 <= lc and w2 <= lc\
  or l1 + w2 <= wc and w1 <= lc and l2 <= lc\
  or w1 + l2 <= wc and l1 <= lc and w2 <= lc\
  or w1 + w2 <= wc and l1 <= lc and l2 <= lc:
    print('YES')
else:
    print('NO')
